### PR TITLE
feat(cardinal): expose close redis method. 

### PR DESCRIPTION
**Pull request by pyrofolium**
Closes: #WORLD-356

## What is the purpose of the change

Expose closing of the client connection during shutdown. 

## Brief Changelog

Added new Close() method to interfaces involving storage. 

## Tests

adjusted tests to use shutdown.

### CODE REVIEW

File: cardinal/server/server_test.go [hunk 1]
```diff
@@ -120,13 +124,8 @@ func setTestTimeout(t *testing.T, timeout time.Duration) {
 func TestHealthEndpoint(t *testing.T) {
 	setTestTimeout(t, 10*time.Second)
 	w := inmem.NewECSWorldForTest(t)
-	sendTx := ecs.NewTransactionType[SendEnergyTx, SendEnergyTxResult]("sendTx")
-	assert.NilError(t, w.RegisterTransactions(sendTx))
-	w.AddSystem(func(world *ecs.World, queue *transaction.TxQueue, _ *log.Logger) error {
-		return nil
-	})
```
Comment (pyrofolium) [inline cardinal/server/server_test.go:127]: Missed removing these in the last PR, removing them here.

File: cardinal/server/server_test.go [hunk 2]
```diff
@@ -177,12 +167,10 @@ func TestShutDownViaMethod(t *testing.T) {
...
 	_, err = http.Get("http://localhost:4040/health")
 	assert.Check(t, err != nil)
-	err = txh.Close()
-	assert.NilError(t, err)
 }
 
 func TestShutDownViaSignal(t *testing.T) {
-	//setTestTimeout(t, 10*time.Second) // If this test is frozen then it failed to shut down, create a failure with panic.
+	setTestTimeout(t, 10*time.Second) // If this test is frozen then it failed to shut down, create a failure with panic.
 	w := inmem.NewECSWorldForTest(t)
```
Comment (pyrofolium) [inline cardinal/server/server_test.go:174]: Same this should be here

File: cardinal/ecs/storage/engine.go [hunk 3]
```diff
@@ -100,3 +100,7 @@ type NonceStorage interface {
 	GetNonce(key string) (uint64, error)
 	SetNonce(key string, nonce uint64) error
 }
+
+type IO interface {
+	Close() error
+}
```
Comment (technicallyty) [inline cardinal/ecs/storage/engine.go:106]: we can just use the interface defined in Go's `io` package: `io.Closer`, i believe it is more standard to do so

Comment (pyrofolium) [reply cardinal/ecs/storage/engine.go]: Did not know this already existed.

**Review by technicallyty — APPROVED** [review]
lgtm

**Comment by technicallyty** [comment]
make sure to update the PR title as well, still has branch name in it

**Pull request merged**